Quotelighter/AnnotationSource.cs

```diff
@@ -67,32 +67,49 @@ private class VerseAnnotationInfo
 
         private readonly IPluginHost m_host;
         private readonly IProject m_project;
+        private readonly ApostropheAnalyzer m_apostropheAnalyzer;
         private readonly HashSet<char> m_allQuoteChars;
 	    private readonly Regex m_findMarksRegex;
 	    private int m_currentBook = -1;
         private readonly Dictionary<int, ChapterAnnotationInfo> m_currentBookAnnotations =
 	        new Dictionary<int, ChapterAnnotationInfo>();
 
-        public AnnotationSource(IPluginHost host, IProject project)
+        public AnnotationSource(IPluginHost host, IProject project, ApostropheAnalyzer apostropheAnalyzer = null)
         {
 	        m_host = host;
 	        m_project = project;
-            project.ProjectDataChanged += ProjectDataChanged;
+	        m_apostropheAnalyzer = apostropheAnalyzer;
+	        project.ProjectDataChanged += ProjectDataChanged;
             project.ScriptureDataChanged += ScriptureDataChanged;
             var quotationMarks = project.Language.QuotationMarkInfo;
             if (quotationMarks?.PrimaryLevels == null)
                 return;
 
             var allMarks = new Dictionary<string, int>();
 
-            StringBuilder bldr = new StringBuilder();
+            var bldr = new StringBuilder();
             int currentLevelInBuilder = -1;
 
+			string EscapeAndAvoidWordMedialApostropheMatching(string mark)
+			{
+				if (mark == "’")
+				{
+					if (m_apostropheAnalyzer == null ||
+					    m_apostropheAnalyzer.TreatSingleClosingCurlyQuoteAsApostrophe == ApostropheAnalyzer.SpecialTreatmentRule.Never)
+					{
+						return mark;
+					}
+					return @"(?<!\p{L}|\p{M})’|’(?!\p{L}|\p{M})";
+				}
+
+				return Regex.Escape(mark);
+			}
+
```

Comment:
Lots of tabs/spaces problems in the files again. You might consider deciding which one you want and changing all the files to that.